migrate.js

```diff
@@ -31,6 +30,30 @@ function makeFileExistsObv(filename) {
   return obv
 }
 
+// Forked from flumecodec because we have to support
+// bendy butt messages which may contain Buffers
+const jsonCodec = {
+  encode: JSON.stringify,
+  decode(str) {
+    const parsed = JSON.parse(str)
+    const content = parsed.value.content
+    if (
+      typeof content === 'object' &&
+      content.type === 'metafeed/add/derived'
+    ) {
```

Comment:
Hopefully this `if` condition will preserve the performance of the jsonCodec since `content.type` is rarely add/derived